inspirehep/modules/workflows/actions/merge_approval.py

```diff
@@ -32,4 +34,16 @@ class MergeApproval(object):
     @staticmethod
     def resolve(obj, *args, **kwargs):
         """Resolve the action taken in the approval action."""
-        pass
+
+        obj.extra_data["approved"] = True
+        obj.extra_data["auto-approved"] = False
+        obj.remove_action()
+        obj.save()
+
+        delayed = True
+        if obj.workflow.name == 'manual_merge':
+            # the manual merge wf should be sync
+            delayed = False
```

Comment:
This is really sad. But I see why it's needed. We need to ensure that whatever happens afterwards in the workflow is peanut and can complete quickly. Otherwise the request will end-up in a Timeout.